crates/wasi-common/yanix/src/file.rs

```diff
@@ -230,20 +233,26 @@ pub unsafe fn fstatat<P: AsRef<OsStr>>(dirfd: RawFd, path: P, flags: AtFlag) ->
 pub unsafe fn fstat(fd: RawFd) -> Result<libc::stat> {
     use std::mem::MaybeUninit;
     let mut filestat = MaybeUninit::<libc::stat>::uninit();
-    Error::from_result(libc::fstat(fd, filestat.as_mut_ptr()))?;
+    from_result(libc::fstat(fd, filestat.as_mut_ptr()))?;
     Ok(filestat.assume_init())
 }
 
 /// `fionread()` function, equivalent to `ioctl(fd, FIONREAD, *bytes)`.
 pub unsafe fn fionread(fd: RawFd) -> Result<u32> {
     let mut nread: libc::c_int = 0;
-    Error::from_result(libc::ioctl(fd, libc::FIONREAD, &mut nread as *mut _))?;
-    Ok(nread.try_into()?)
+    from_result(libc::ioctl(fd, libc::FIONREAD, &mut nread as *mut _))?;
+    // FIONREAD returns a non-negative int if it doesn't fail, or it'll fit in a u32 if it does.
+    //
+    // For the future, if we want to be super cautious and avoid assuming int is 32-bit, we could
+    // widen fionread's return type here, since the one place that calls it wants a u64 anyway.
+    Ok(nread.try_into().unwrap())
 }
 
 /// This function is unsafe because it operates on a raw file descriptor.
 /// It's provided, because std::io::Seek requires a mutable borrow.
 pub unsafe fn tell(fd: RawFd) -> Result<u64> {
-    let offset: i64 = Error::from_result(libc::lseek(fd, 0, libc::SEEK_CUR))?;
-    Ok(offset.try_into()?)
+    let offset: i64 = from_result(libc::lseek(fd, 0, libc::SEEK_CUR))?;
```

Comment:
Wanna go ahead and fix this for other platforms and remove the `: i64` here?

Reply:
Ah, you're probably referring to #1231. Good call, lemme do just that!

Reply:
Done in ed98a80.